src/libraries/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -915,22 +949,14 @@ public static Task<string[]> ReadAllLinesAsync(string path, Encoding encoding, C
 
         private static async Task<string[]> InternalReadAllLinesAsync(string path, Encoding encoding, CancellationToken cancellationToken)
         {
-            Debug.Assert(!string.IsNullOrEmpty(path));
-            Debug.Assert(encoding != null);
+            List<string> lines = new List<string>();
 
-            using (StreamReader sr = AsyncStreamReader(path, encoding))
+            await foreach (string line in InternalReadLinesAsync(path, encoding, cancellationToken).ConfigureAwait(false))
```

Comment:
While the reuse is nice, this is going to make File.ReadAllLinesAsync more expensive, though I don't know how measurable or impactful it'll be.  Since it's really not saving much code (especially if/when ReadLineAsync takes a CancellationToken), we may want to just leave it the way it was.

Reply:
Removed 👍🏻